### PR TITLE
Add CloudTasks queue and task template

### PR DESCRIPTION
**Pull request by sourced-praveenc**

### CODE REVIEW

File: community/cloud-foundation/templates/cloud_tasks/task.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    for prop in optional_properties:
+        if prop in properties['task']:
+            task['properties']['task'][prop] = properties['task'][prop]
+
+    resources.append(task)
+
+    # Output variables
+    output_props = [
+        'name'
+        # 'name',
```
Comment (tiadobatima1) [inline community/cloud-foundation/templates/cloud_tasks/task.py]: cleanup these comments?

File: community/cloud-foundation/templates/cloud_tasks/examples/cloud_tasks_queue.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+resources:
+  - name: mysupertestqueue
+    type: queue.py
+    properties:
+      rateLimits:
+        maxDispatchesPerSecond: 10
+        maxConcurrentDispatches: 5
+      retryConfig:
+        maxAttempts: 2
+        maxRetryDuration: "60s"
```
Comment (tiadobatima1) [inline community/cloud-foundation/templates/cloud_tasks/examples/cloud_tasks_queue.yaml]: This is a string. No need for quotes

File: community/cloud-foundation/templates/cloud_tasks/examples/cloud_tasks_queue.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        minBackoff: "10s"
```
Comment (tiadobatima1) [inline community/cloud-foundation/templates/cloud_tasks/examples/cloud_tasks_queue.yaml]: Maybe I'm wrong, but according to the convoluted API doc, `minBackoff` and `maxBackoff` should be in seconds, so "60s" and "120s" should fail?

File: community/cloud-foundation/templates/cloud_tasks/queue.py.schema [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+        type: string
+        description: |
+          A task will be scheduled for retry between minBackoff and maxBackoff
+          duration after it fails, if the queue's RetryConfig specifies that
+          the task should be retried. If unspecified when the queue is created,
+          Cloud Tasks will pick the default. minBackoff will be truncated to
+          the nearest second. This field has the same meaning as 
+          min_backoff_seconds in queue.yaml/xml.
+          https://cloud.google.com/appengine/docs/standard/python/config/queueref#retry_parameters
+          A duration in seconds with up to nine fractional digits, terminated
```
Comment (tiadobatima1) [inline community/cloud-foundation/templates/cloud_tasks/queue.py.schema]: API doc doesn't really say anything about this being a string terminated is "s". I haven't tested, so I don't know for sure.

File: community/cloud-foundation/templates/cloud_tasks/queue.py.schema [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        description: |
+          A task will be scheduled for retry between minBackoff and maxBackoff
+          duration after it fails, if the queue's RetryConfig specifies that
+          the task should be retried. If unspecified when the queue is created,
+          Cloud Tasks will pick the default. maxBackoff will be truncated to
+          the nearest second. This field has the same meaning as
+          max_backoff_seconds in queue.yaml/xml.
+          A duration in seconds with up to nine fractional digits, terminated
+          https://cloud.google.com/appengine/docs/standard/python/config/queueref#retry_parameters
+          by 's'. For example: "3.5s".
```
Comment (tiadobatima1) [inline community/cloud-foundation/templates/cloud_tasks/queue.py.schema]: API doc doesn't really say anything about this being a string terminated is "s". I haven't tested, so I don't know for sure

File: community/cloud-foundation/templates/cloud_tasks/queue.py.schema [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+        description: |
+          A task will be scheduled for retry between minBackoff and maxBackoff
+          duration after it fails, if the queue's RetryConfig specifies that
+          the task should be retried. If unspecified when the queue is created,
+          Cloud Tasks will pick the default. maxBackoff will be truncated to 
+          the nearest second. This field has the same meaning as 
+          max_backoff_seconds in queue.yaml/xml.
+          https://cloud.google.com/appengine/docs/standard/python/config/queueref#retry_parameters
+          A duration in seconds with up to nine fractional digits, terminated
+          by 's'. For example: "3.5s".
```
Comment (tiadobatima1) [inline community/cloud-foundation/templates/cloud_tasks/queue.py.schema]: API doc doesn't really say anything about this being a string terminated is "s". I haven't tested, so I don't know for sure.

File: community/cloud-foundation/templates/cloud_tasks/examples/cloud_tasks_queue.yaml (hunk 3 above)
Comment (sourced-praveenc) [reply community/cloud-foundation/templates/cloud_tasks/examples/cloud_tasks_queue.yaml]: Check this, it does say that it should terminate with "s"
https://cloud.google.com/tasks/docs/reference/rest/v2beta3/projects.locations.queues#RateLimits

File: community/cloud-foundation/templates/cloud_tasks/queue.py.schema (hunk 4 above)
Comment (sourced-praveenc) [reply community/cloud-foundation/templates/cloud_tasks/queue.py.schema]: Check this, it does say that it should terminate with "s"
https://cloud.google.com/tasks/docs/reference/rest/v2beta3/projects.locations.queues#RateLimits

File: community/cloud-foundation/templates/cloud_tasks/queue.py.schema (hunk 5 above)
Comment (sourced-praveenc) [reply community/cloud-foundation/templates/cloud_tasks/queue.py.schema]: Check this, it does say that it should terminate with "s"
https://cloud.google.com/tasks/docs/reference/rest/v2beta3/projects.locations.queues#RateLimits

File: community/cloud-foundation/templates/cloud_tasks/queue.py.schema (hunk 6 above)
Comment (sourced-praveenc) [reply community/cloud-foundation/templates/cloud_tasks/queue.py.schema]: Check this, it does say that it should terminate with "s"
https://cloud.google.com/tasks/docs/reference/rest/v2beta3/projects.locations.queues#RateLimits

File: community/cloud-foundation/templates/cloud_tasks/examples/cloud_tasks_queue.yaml (hunk 3 above)
Comment (tiadobatima1) [reply community/cloud-foundation/templates/cloud_tasks/examples/cloud_tasks_queue.yaml]: I looked at another doc and it doesn't show that. LOL 😂 
https://developers.google.com/apis-explorer/#p/cloudtasks/v2beta3/cloudtasks.projects.locations.queues.create

**Pull request closed without merging**